packages/cli/src/commands/screenshot-diff/screenshot-diff.command.ts

```diff
@@ -18,199 +24,244 @@ import {
-        headReplayId,
-        screenshotFileName,
-        diff: comparisonResult.diff,
-      });
+          return [
+            {
+              identifier,
+              outcome:
+                comparisonResult.mismatchFraction > diffThreshold
+                  ? "diff"
+                  : "no-diff",
+              headScreenshotFile: `screenshots/${screenshotFileName}`,
+              baseScreenshotFile: `screenshots/${screenshotFileName}`,
+              width: baseScreenshot.width,
+              height: baseScreenshot.height,
+              mismatchPixels: comparisonResult.mismatchPixels,
+              mismatchFraction: comparisonResult.mismatchFraction,
+            },
+          ];
+        }
+      )
+    )
+  ).flat();
 
-      comparisonResults.push({
-        baseScreenshotFile,
-        headScreenshotFile,
-        comparisonResult,
-        outcome:
-          comparisonResult.mismatchFraction > diffThreshold ? "fail" : "pass",
-      });
-    }
+  const diffUrl = await getDiffUrl(client, baseReplayId, headReplayId);
+  logger.info(`View screenshot diff at ${diffUrl}`);
 
-    await postScreenshotDiffStats(client, {
+  return [...missingHeadImagesResults, ...headDiffResults];
+};
+
+export const checkScreenshotDiffResult = ({
+  baseReplayId,
+  headReplayId,
+  results,
+  diffOptions,
+}: {
+  baseReplayId: string;
+  headReplayId: string;
+  results: ScreenshotDiffResult[];
+  diffOptions: ScreenshotDiffOptions;
+}): void => {
+  const logger = log.getLogger(METICULOUS_LOGGER_NAME);
+
+  const missingHeadImagesResults = results.filter(
+    ({ outcome }) => outcome === "missing-head"
+  ) as ScreenshotDiffResultMissingHead[];
+  if (missingHeadImagesResults.length) {
+    const message = `Head replay is missing screenshots: ${missingHeadImagesResults
+      .map(({ baseScreenshotFile }) => baseScreenshotFile)
+      .sort()} => FAIL!`;
+    logger.info(message);
+    throw new ScreenshotDiffError(message, {
       baseReplayId,
       headReplayId,
-      stats: {
-        width: 0,
-        height: 0,
-        mismatchPixels: totalMismatchPixels,
-      },
     });
+  }
 
-    const diffUrl = await getDiffUrl(client, baseReplayId, headReplayId);
-    logger.info(`View screenshot diff at ${diffUrl}`);
-
-    comparisonResults.forEach((result) => {
-      logComparisonResultMessage(
-        logger,
-        `${Math.round(
-          result.comparisonResult.mismatchFraction * 100
-        )}% pixel mismatch for screenshot ${basename(
-          result.headScreenshotFile
-        )} (threshold is ${Math.round(diffThreshold * 100)}%)`,
-        result.outcome
-      );
-    });
+  const missingBaseImagesResults = results.filter(
+    ({ outcome }) => outcome === "missing-base"
+  ) as ScreenshotDiffResultMissingBase[];
+  if (missingHeadImagesResults.length) {
+    const message = `Notice: Base replay is missing screenshots: ${missingBaseImagesResults
+      .map(({ headScreenshotFile }) => basename(headScreenshotFile))
```

Comment:
Note: we use basename here but not in the missing-head case

Reply:
Done in c07f118